plugins/openai/api_key.go

```diff
@@ -0,0 +1,39 @@
+package openai
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://beta.openai.com/docs"),
+		ManagementURL: sdk.URL("https://beta.openai.com/account/api-keys"),
```

Comment:
Given these URLs start with `beta`, I'm assuming this product is still in beta, although I was not able to find a beta notice on the site.

I just wanted to see how exactly would a potential protocol change, on the side of a third party CLI, play in with our backwards compatibility promise. I'd reckon our promise does not extend to the third party tools we support through our plugins, but wanted to raise this just so we can make a conscious  decision on it. cc @SimonBarendse

Reply:
Great question!

We'll want to implement versioning to be able to handle backwards incompatible changes from the third-party CLIs. The plugin could behave differently depending on the version of the third-party CLI you have installed.